addons/common/functions/fnc_initRagdoll.sqf

```diff
@@ -25,42 +25,20 @@ GVAR(ragdollRunning) = true;
 
 ["ace_unconscious", {
     params [["_unit", objNull],["_state", false]];
-    if(_unit != player || {!GVAR(ragdolling)}) exitWith {}; // only ragdoll players and only ragdolling if it active
-    if(_state && {vehicle _unit == _unit && {!([_unit] call ace_medical_fnc_isBeingCarried) && {!([_unit] call ace_medical_fnc_isBeingDragged)}}}) then {
-        // ragdoll player
+    if(!GVAR(ragdolling)) exitWith {}; // ragdolling if it's active
+    if((!(isPlayer _unit) && {!GVAR(ragdoll_ai)})) exitWith {}; // only ragdoll players and only ragdolling AI if that option is set active
+    if(_state && {(isNull objectParent _unit) && {!([_unit] call ace_medical_fnc_isBeingCarried) && {!([_unit] call ace_medical_fnc_isBeingDragged)}}}) then {
+        // ragdoll unit
         _unit setUnconscious true;
     };
     if(!_state) then {
-        // player woke up before ragdolling was finished
+        // unit woke up before ragdolling was finished
         _unit setUnconscious false;
     };
 }] call CBA_fnc_addEventHandler;
 
-player addEventHandler ["AnimChanged", {
-    params ["_unit","_anim"];
-    if(!GVAR(ragdolling)) exitWith {}; // disable ragdolling mid mission
-    if(_anim == "unconsciousrevivedefault" && {alive _unit && {_unit getVariable ["ACE_isUnconscious",false] && {vehicle _unit == _unit}}}) then {
-        // unit stopped ragdolling, apply ace_death animation to unit
-        _unit setUnconscious false;
-        [_unit, [_unit] call ace_common_fnc_getDeathAnim, 2, true] call ace_common_fnc_doAnimation;
-        if(isMultiplayer) then {
-            // combat sync issues
-            [
-                {
-                    params ["_unit"];
-                    if(!(_unit getVariable ["ACE_isUnconscious",false])) then {
-                        // unit is not unconscious anymore
-                        _unit setUnconscious false;
-                    } else {
-                        // unit is still unconscious, reapply death animation just in case and sync it again to all clients
-                        [_unit, [_unit] call ace_common_fnc_getDeathAnim, 2, true] call ace_common_fnc_doAnimation;
-                    };
-                }, // code
-                [_unit], // params
-                5 // delay
-            ] call CBA_fnc_waitAndExecute;
-        };
-    };
-}];
+["CAManBase", "AnimChanged", {
+    _this call FUNC(ragdoll_animChangedEH);
```

Comment:
what's the thought behind local on this as animchanged is global EH

Reply:
In the end ragdolls are different on each machine and the ragdolling phase takes a shorter or longer time on each machine. I prefer to apply them locally only for AI so that a hopefully fitting animation can be chosen.
For players the EH does not fire all the time, for some reason, that is why for a player the player themselves will tell everyone on the server that they assumed a certain position.